subsys/usb/class/cdc_acm.c

```diff
@@ -301,6 +301,18 @@ static void tx_work_handler(struct k_work *work)
 	len = ring_buf_get_claim(dev_data->tx_ringbuf, &data,
 				 CONFIG_USB_CDC_ACM_RINGBUF_SIZE);
 
+	/*
+	 * Send less data to avoid having to send a zero-length packet. When
+	 * send a zero-length packet, the host or the application running on
+	 * the host may believe that there is no more data to be received
+	 * (i.e. the transaction has completed), hence not triggering another
+	 * IN transaction, and thus preventing any more data to be sent from
+	 * Zephyr to the Host.
+	 */
+	if (!(len % CONFIG_CDC_ACM_BULK_EP_MPS)) {
```

Comment:
I think `len` should be checked for zero.

Reply:
I don't think this could actually happen, but I agree it's better to be safe than sorry. However should we just return from this function if it happens? From what I understand calling usb_transfer() with a length of 0 will send a ZLP, which is what we want to avoid. Note that this problem is not introduced by this PR, it was there also in the original code.

Reply:
I think it should just return if len == 0. Agree, the issue was already there but you patch escalate it a little :-)